servers/physics_3d/godot_area_3d.cpp

```diff
@@ -327,9 +332,14 @@ void GodotArea3D::call_queries() {
 }
 
 void GodotArea3D::compute_gravity(const Vector3 &p_position, Vector3 &r_gravity) const {
+	Vector3 gravity_vector = get_gravity_vector();
+	if (gravity_apply_transform) {
+		gravity_vector = get_transform().xform(gravity_vector);
```

Comment:
The transformation here is different, when a point it should transform the point to local space, but when a direction it should just rotate and scale, not translate, i.e. transform by the basis only

This will make the gravity stronger and stronger the further from origin the area is

Reply:
I would say the code for point gravity should be unchanged and the other case should be changed to:
```cpp
	else if (gravity_apply_transform) {
		r_gravity = get_transform().get_basis().xform(get_gravity_vector()) * get_gravity();
	} else {
		r_gravity = get_gravity_vector() * get_gravity();
	}
```

Or similar

Reply:
Okay I've done it like you suggested. I agree that this way is better and now this should no longer break compatibility.